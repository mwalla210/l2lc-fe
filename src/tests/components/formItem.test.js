import React from 'react'
import renderer from 'react-test-renderer'
import FormItem from '../../components/formItem'

const defaultOptions = {
  isValid: true,
  errorText: 'errorText',
  label: 'label',
  required: true,
  disabled: false,
}

const alternateOptions = {
  isValid: true,
  errorText: 'errorText',
  label: 'label',
  required: true,
  disabled: true,
}

describe('FormItem', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <FormItem {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders with snapshot, alternate options', () => {
    let options = Object.assign({}, alternateOptions)
    const component = renderer.create(
      <FormItem {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
