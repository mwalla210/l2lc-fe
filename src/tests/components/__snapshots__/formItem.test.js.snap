// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem Renders with snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="alert alert-warning"
    role="alert"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      style={
        Object {
          "color": "orange",
        }
      }
    >
      <strong>
        Warning!
      </strong>
       errorText
    </div>
  </div>
  <label>
    label
  </label>
   
  <span
    style={
      Object {
        "color": "orange",
      }
    }
  >
     *
  </span>
</div>
`;

exports[`FormItem Renders with snapshot, alternate options 1`] = `
<div
  className="form-group"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="alert alert-warning"
    role="alert"
    style={
      Object {
        "color": "orange",
      }
    }
  >
    <div
      style={
        Object {
          "color": "orange",
        }
      }
    >
      <strong>
        Warning!
      </strong>
       errorText
    </div>
  </div>
  <label>
    label
  </label>
   
</div>
`;
